cartridges/int_algolia_sfra/cartridge/static/default/js/algolia/insights-config.js

```diff
@@ -138,19 +119,18 @@ function enableInsights(appId, searchApiKey, productsIndex) {
     // https://github.com/SalesforceCommerceCloud/storefront-reference-architecture/blob/dec9c7c684275127338ac3197dfaf8fe656bb2b7/cartridges/app_storefront_base/cartridge/controllers/Order.js#L87
     // Our 'algolia-insights' tag exposes its "data-order" attribute only in that case. When it is present, we can send the purchase event.
 
-    const order = insightsData && insightsData.dataset.order;
-    if (order) {
-        const orderObj = JSON.parse(order);
-        const products = orderObj.items.items.slice(0, 20); // The Insights API accepts up to 20 objectID
-
+    const algoliaObj = insightsData && insightsData.dataset.algoliaobj;
+    if (algoliaObj) {
+        const productsObj = JSON.parse(algoliaObj);
+        const products = productsObj.items.slice(0, 20); // The Insights API accepts up to 20 objectID
         const objectIDs = [];
         const objectData = [];
-        let currency;
+        let currency = productsObj.currency;
 
         products.forEach((product) => {
             const productInfo = {
                 price: product.price.sales.value,
-                quantity: product.quantity,
+                quantity: product.qty,
             };
             if (product.price.list) {
```

Comment:
This logic is now duplicated since the discount is already calculated on the backend.